engine/src/main/java/org/terasology/entitySystem/entity/EntityManager.java

```diff
@@ -18,88 +18,22 @@
 import org.terasology.entitySystem.Component;
 import org.terasology.entitySystem.event.internal.EventSystem;
 import org.terasology.entitySystem.metadata.ComponentLibrary;
-import org.terasology.entitySystem.prefab.Prefab;
 import org.terasology.entitySystem.prefab.PrefabManager;
-import org.terasology.math.geom.Quat4f;
-import org.terasology.math.geom.Vector3f;
 
 import java.util.Map;
 
 /**
  */
-public interface EntityManager {
+public interface EntityManager extends EntityCache {
 
     /**
-     * Creates an EntityBuilder.
+     * Creates a new EntityRef in sector-scope
      *
-     * @return A new entity builder
+     * @return the newly created EntityRef
      */
-    EntityBuilder newBuilder();
-
-    /**
-     * Creates an EntityBuilder, from a prefab
-     *
-     * @return A new entity builder
-     */
-    EntityBuilder newBuilder(String prefabName);
-
-    /**
-     * Creates an EntityBuilder, from a prefab
-     *
-     * @return A new entity builder
-     */
-    EntityBuilder newBuilder(Prefab prefab);
-
-    /**
-     * @return A references to a new, unused entity
-     */
-    EntityRef create();
-
-    /**
-     * @return A references to a new, unused entity with the desired components
-     */
-    EntityRef create(Component... components);
-
-    /**
-     * @return A references to a new, unused entity with the desired components
-     */
-    EntityRef create(Iterable<Component> components);
-
-    /**
-     * @param prefabName The name of the prefab to create.
-     * @return A new entity, based on the the prefab of the given name. If the prefab doesn't exist, just a new entity.
-     */
-    EntityRef create(String prefabName);
-
-    /**
-     * @param prefab
-     * @return A new entity, based on the given prefab
-     */
-    EntityRef create(Prefab prefab);
-
-    // TODO: Review. Probably better to move these into a static helper
-
-    /**
-     * @param prefab
-     * @param position
-     * @return A new entity, based on the given prefab, at the desired position
-     */
-    EntityRef create(String prefab, Vector3f position);
-
-    /**
-     * @param prefab
-     * @param position
-     * @return A new entity, based on the given prefab, at the desired position
-     */
-    EntityRef create(Prefab prefab, Vector3f position);
-
-    /**
-     * @param prefab
-     * @param position
-     * @param rotation
-     * @return
-     */
-    EntityRef create(Prefab prefab, Vector3f position, Quat4f rotation);
+    default EntityRef createSectorEntity() {
```

Comment:
Yep, you're right. `EntityRef.NULL` is the preferred way of returning a null entity, as it behaves sensibly by avoiding NullPointerExceptions.

I'm actually not sure why I implemented this as a default method; the only EntityManager in the engine is the Pojo one, for which I've implemented the method anyway (I expect my thinking was that I didn't want to have to go and implement it in multiple places while the interface is still changing). It's probably best to just remove the default implementation and leave it abstract.